### ISSUE TITLE
Requirements for the high-level wrapper

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/r2dii.match#45]
Follows https://github.com/2DegreesInvesting/r2dii.match/issues/33#issuecomment-565370557

The required interface is higher level than what we currently have. We can name the top level wrapper with a name that makes more sense to users/analysts, and the all lower level functions with a name that makes more sense for developers. 

At the top level, users need to know how the function solves their problem, using their "domain specific language".

At the low level, I need to read the function name and fairly guess what it does without reading its body (i.e. Does it add, or overwrite a column? Does it join two datasets? Does it output a modified version of the input?). The language here is specific to the developing domain. 

For example, if I am reading code at the low level (internal functions), when I read "match" I expect something similar to `base::match()`, not similar to `base::merge()`, nor `dplyr::left_join()`, nor `stringdist::stringsim()`.

But to make this distinction (high vs. low level) clear, low level functions should be internal. 

My plans as I see them now:

* Make almost all current functions internal, and rename them closer to the developer-domain.
* Add a new top level wrapper and name it close to the user/analyst domain.

--

The requirements Klaus listed are roughly these:

* Rename:

> Maybe better to call this function match_2_datasets_via_simpler_name (the function is also not limited to ALD <-> loanbook)

* Add more arguments:

> The first wrapper-function should have the inputs: loanbook data, ald, grouping variable (sector, potentially others), confidence threshold (default should be higher 0.5 will create very very weird matches)

* Output original columns:

> For the manual verification output (csv) we need to add the original names back in and potentially all other relevant information that makes it much easier and user-friendly than just showing the simpler names.

* Allow grouping:

> One enhancement I see we can & should do: Would be good to write the function in a way that allows “grouping” by another variable such as country_of_domicile – next version.

**Comment by maurolepore** [comment]
Here is my first attempt of such a top-level wrapper:

``` r
library(dplyr)  # Use tibble()
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(r2dii.match)

x <- tibble(sector = c("A", "B", "B"), simpler_name = c("xa", "xb", "xc"))
y <- tibble(sector = c("A", "B", "C"), simpler_name = c("ya", "yb", "yc"))

match_by_simpler_name(x, y, threshold = 0.5)
#> # A tibble: 3 x 5
#>   simpler_name_x simpler_name_y score sector_x sector_y
#>   <chr>          <chr>          <dbl> <chr>    <chr>   
#> 1 xa             ya             0.667 A        A       
#> 2 xb             yb             0.667 B        B       
#> 3 xc             yc             0.667 B        C
```

<sup>Created on 2019-12-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

**Comment by maurolepore** [comment]
However, the top-level wrapper could also "prepare" the data so the input would not have a `simpler_name` column and thus it may be confusing to have "simpler_name"  in the name of the top level wrapper (it would be unclear where "simpler_name" comes from).

The interface might look like this:

```r
# Prepares the data (i.e. creates `simpler_name`), then does scores `simpler_name`
score_name(loanbook, ald)
```

Then other conceivable names might include: `score_join_by_name()`, `add_and_score_simpler_name()`, `score_name_match()`

**Comment by maurolepore** [comment]
Closed along with #47
